signature/types.go

```diff
@@ -182,15 +186,31 @@ type SignerInfo struct {
 	Signature []byte
 }
 
-// Payload represents payload in bytes and its content type.
+// Payload is the payload to be signed.
+//
+// When CoseHashEnvelopePayload is provided, ContentType and Content are ignored.
+// In this case, the payload represents a COSE Hash Envelope payload.
+// Reference: https://www.ietf.org/archive/id/draft-ietf-cose-hash-envelope-05.html
 type Payload struct {
 	// ContentType specifies the content type of payload.
+	//
+	// It is ignored, when [Payload.CoseHashEnvelopePayload] is present.
 	ContentType string
 
 	// Content contains the raw bytes of the payload.
 	//
 	// For JWS envelope, Content is limited to be JSON format.
+	//
+	// It is ignored, when [Payload.CoseHashEnvelopePayload] is present.
 	Content []byte
+
+	// CoseHashEnvelopePayload is the payload to be signed when signing
+	// under the COSE format and COSE Hash Envelope is required as result.
+	//
+	// When present, [Payload.ContentType] and [Payload.Content] are ignored.
+	//
+	// Reference: https://www.ietf.org/archive/id/draft-ietf-cose-hash-envelope-05.html
+	CoseHashEnvelopePayload *cose.HashEnvelopePayload
```

Comment:
nit: code style: naming convention
```suggestion
	COSEHashEnvelopePayload *cose.HashEnvelopePayload
```

Reply:
Updated.